router/batchrouter/asyncdestinationmanager/eloqua/utils.go

```diff
@@ -0,0 +1,195 @@
+	var line string
+	fmt.Print(scanner.Err())
+	if scanner.Scan() {
+		line = scanner.Text()
+	}
+	var data TransformedData
+	if err := json.Unmarshal([]byte(line), &data); err != nil {
+		fmt.Println("Error in unmarshalling data")
+	}
+	return getKeys(data.Message.Data)
+}
+func createCSVFile(fields []string, file *os.File) (string, error) {
+	file.Seek(0, 0)
+	scanner := bufio.NewScanner(file)
+	scanner.Buffer(nil, 50000*1024)
+	localTmpDirName := fmt.Sprintf(`/%s/`, misc.RudderAsyncDestinationLogs)
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		return "", err
+	}
+	path := path.Join(tmpDirPath, localTmpDirName, uuid.NewString())
+	csvFilePath := fmt.Sprintf(`%v.csv`, path)
+	csvFile, err := os.Create(csvFilePath)
+	csvWriter := csv.NewWriter(csvFile)
+	csvWriter.Write(fields)
+	csvWriter.Flush()
+	var line string
+	for scanner.Scan() {
+		line = scanner.Text()
+		var data TransformedData
+		if err := json.Unmarshal([]byte(line), &data); err != nil {
+			fmt.Println("Error in unmarshalling data")
+		}
+
+		var values []string
+		for _, field := range fields {
+			values = append(values, data.Message.Data[field].(string))
+		}
+		csvWriter.Write(values)
+	}
+
+	csvWriter.Flush()
+	return csvFilePath, nil
+}
+
+func createBodyForImportDefinition(evenType string, fields []string, eloquaFields *Fields, file *os.File) (map[string]interface{}, error) {
+	file.Seek(0, 0)
+	scanner := bufio.NewScanner(file)
+	scanner.Buffer(nil, 50000*1024)
+	data := TransformedData{}
+	if scanner.Scan() {
+		line := scanner.Text()
+		if err := json.Unmarshal([]byte(line), &data); err != nil {
+			return nil, fmt.Errorf("error in unmarshalling data while creating body for import definition. %v", err)
+		}
+	}
+
+	if evenType == "identify" {
+		var contactFieldStatement = make(map[string]string)
+		for _, val := range fields {
+			for _, val1 := range eloquaFields.Items {
+				if val1.InternalName == val {
+					contactFieldStatement[val] = val1.Statement
+				}
+			}
+		}
+		return map[string]interface{}{
+			"name":                    "Rudderstack-Contact-Import",
+			"fields":                  contactFieldStatement,
+			"identifierFieldName":     data.Message.IdentifierFieldName,
+			"isSyncTriggeredOnImport": false,
+		}, nil
+	} else if evenType == "track" {
+		var customObjectFieldStatement = make(map[string]string)
+		for _, val := range fields {
+			for _, val1 := range eloquaFields.Items {
+				if val1.InternalName == val {
+					customObjectFieldStatement[val] = val1.Statement
+				}
+			}
+		}
+		return map[string]interface{}{
+			"name":                    "Rudderstack-CustomObject-Import",
+			"updateRule":              "always",
+			"fields":                  customObjectFieldStatement,
+			"identifierFieldName":     data.Message.MapDataCardsSourceField,
+			"isSyncTriggeredOnImport": false,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("unable to create body for import definition")
+}
+
+func difference(a, b []int64) []int64 {
```

Comment:
can you check samber/lo package might have util for this:
https://github.com/samber/lo#difference